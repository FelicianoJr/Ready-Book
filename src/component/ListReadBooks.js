import React from 'react'
import { Link } from 'react-router-dom'
import BookShelf from './BookShelf'

const ListReadBooks = ({ books = [], update, titles = [], clearBook }) => {

    return (
        <div className="list-books">
            <div className="list-books-title">
                <h1>MyReads</h1>
            </div>
            <div className="list-books-content">
                <div>
                    {titles.map(title => {
                        return <BookShelf key={title} books={books}
                            title={title} update={update} />
                    })}
                </div>
            </div>
            <div className="open-search">
                <Link to="/search" onClick={() => clearBook()} ></Link>
            </div>
        </div>
    )
};

export default ListReadBooks;