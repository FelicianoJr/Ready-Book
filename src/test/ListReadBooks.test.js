import ListReadBooks from '../component/ListReadBooks'
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Link } from 'react-router-dom';

describe("<ListReadBooks/>", () => {

    let update;
    let props;
    let titles = ['read', 'wantToRead', 'continueRead'];
    let clearBook;

    beforeEach(() => {
        update, clearBook = jest.fn();
        props = {
            books: [{
                id: 0, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React', shelf: 'read'
            },
            {
                id: 1, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React Native', shelf: 'read'
            }],
            update: update,
            titles: titles,
            clearBook: clearBook
        };
    })

    it("should render Link /search", () => {
        const wrapper = shallow(<ListReadBooks {...props} />)
        expect(wrapper.find(Link).props().to).toEqual('/search');
    });

    it('should render MyReads in h1', () => {
        const wrapper = shallow(<ListReadBooks {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('h1').text()).toEqual('MyReads');
    });

    it('should render five <div>', () => {
        const wrapper = shallow(<ListReadBooks {...props} />)
        expect(wrapper.find('div').length).toEqual(5);
    })

})