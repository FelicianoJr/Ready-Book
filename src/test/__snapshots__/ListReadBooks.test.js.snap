exports[`<ListReadBooks/> should render MyReads in h1 1`] = `
<div
  className="list-books">
  <div
    className="list-books-title">
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content">
    <div>
      <BookShelf
        books={
          Array [
            Object {
              "id": 0,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React",
            },
            Object {
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React Native",
            },
          ]
        }
        key="read"
        title="read" />
      <BookShelf
        books={
          Array [
            Object {
              "id": 0,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React",
            },
            Object {
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React Native",
            },
          ]
        }
        key="wantToRead"
        title="wantToRead" />
      <BookShelf
        books={
          Array [
            Object {
              "id": 0,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React",
            },
            Object {
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React Native",
            },
          ]
        }
        key="continueRead"
        title="continueRead" />
    </div>
  </div>
  <div
    className="open-search">
    <Link
      onClick={[Function]}
      replace={false}
      to="/search" />
  </div>
</div>
`;
